src/collections/Form/Form.js

```diff
@@ -94,10 +94,14 @@ class Form extends Component {
   static Select = FormSelect
   static TextArea = FormTextArea
 
-  handleSubmit = e => {
-    const { action, onSubmit } = this.props
-    if (!action) e.preventDefault()
-    if (onSubmit) onSubmit(e, { ...this.props })
+  handleSubmit = (...args) => {
+    const { action } = this.props
+    const [e] = args
```

Comment:
Hm, I see the issue.  I don't like that this makes the signature awkward as the position of the `props` argument can never be known for certain anymore.  It depends on how many arguments the underlying component passed back.  I'd like to keep it `(e, data)` for consistency with all our signatures.

That said, perhaps we should extend out signatures to pass any other args _after_ these.  So:

```
handleSubmit = (e, ...args) => {
  const { action } = this.props
  if (!action) _.invoke(e, 'preventDefault')
  _.invoke(this.props, 'onSubmit', e, this.props, ...args)
}
```

This way our signature is `(e, data, ...args)`.  Thoughts?